pwiz_tools/Skyline/Controls/Graphs/GraphChromatogram.cs

```diff
@@ -2292,6 +2297,43 @@ private void SetRetentionTimeIndicators(ChromGraphItem chromGraphPrimary,
                 nodePeps, lookupSequence, lookupMods);
             SetRetentionTimeIdIndicators(chromGraphPrimary, settings,
                 nodeGroups, lookupSequence, lookupMods);
+
+            if (!settings.PeptideSettings.Libraries.IsLoaded)
+            {
+                // No reason to continue if the Library is still loading or exception will occur in call to TryGetSpectralLibraryIonMobilities
+                return;
+            }
+
+            for (var index = 0; index < nodeGroups.Length; index++)
+            {
+                var libKey = nodeGroups[index].GetLibKey(settings, nodePeps[0]);
+                LibraryIonMobilityInfo libImInfo;
+
+                if (settings.PeptideSettings.Libraries.TryGetSpectralLibraryIonMobilities(new[] { libKey },
+                        chromGraphPrimary.Chromatogram.FilePath, out libImInfo))
+                {
+                    if (libImInfo.GetLibraryMeasuredCollisionalCrossSection(libKey) != null)
+                    {
+                        if (chromGraphPrimary.CCSPredictions == null)
+                        {
+                            chromGraphPrimary.CCSPredictions = new [] {                           
+                                new ChromGraphItem.ChargeAndCcsIndex(0,
+                                    new ChromGraphItem.ChargeAndCollisionalCrossSection(
+                                        nodeGroups[index].PrecursorCharge,
+                                        libImInfo.GetLibraryMeasuredCollisionalCrossSection(libKey)))
+                            };
+                        }
+                        else
+                        {
+                            chromGraphPrimary.CCSPredictions[chromGraphPrimary.CCSPredictions.Count] =

```

Comment:
`chromGraphPrimary.CCSPredictions[chromGraphPrimary.CCSPredictions.Count]` is guaranteed to throw an index out of range exception.
The simplest would be to make the "CCSPredictions" ChromGraphItem be a `List<ChargeAndCcsIndex>` and call the Add method on the List.